### PR TITLE
Change namespace of ILogger to Core

### PR DESCRIPTION
**Pull request by bigfluffycookie**
fixes https://github.com/SonarSource/sonarlint-visualstudio/issues/4046

### CODE REVIEW

File: src/CFamily.UnitTests/CMake/CompilationDatabaseLocatorTests.cs [hunk 1]
```diff
@@ -27,8 +27,6 @@
 using SonarLint.VisualStudio.CFamily.CMake;
 using SonarLint.VisualStudio.Core;
 using SonarLint.VisualStudio.Core.CFamily;
-using SonarLint.VisualStudio.Infrastructure.VS;
```
Comment (bigfluffycookie) [inline src/CFamily.UnitTests/CMake/CompilationDatabaseLocatorTests.cs:30]: I originally removed all unnused references and tried to put back the ones that are not relevant to this pr. However, I definetely missed a few but it should not really matter.

**Comment by bigfluffycookie** [comment]
This was not fun at all :')

**Comment by duncanp-sonar** [comment]
> This was not fun at all :')

239 changed files - must be nearly a record for a single PR!

**Review by duncanp-sonar — APPROVED** [review]
Nice!

**Pull request merged**